File: minesweeper/java-minesweeper/src/main/java/it/chicio/minesweeper/field/parser/FieldsParsingStatusBuilder.java
package it.chicio.minesweeper.field.parser;

import it.chicio.minesweeper.field.Field;

import java.util.ArrayList;

public class FieldsParsingStatusBuilder {
    private Field currentField;
    private String currentRowContent;
    private int currentRow;
    private int headerNumberOfRowsForCurrentField;
    private ArrayList<Field> fieldsParsed;

    FieldsParsingStatusBuilder withCurrentField(Field currentField) {
        this.currentField = currentField;
        return this;
    }

    FieldsParsingStatusBuilder withCurrentRowContent(String currentRowContent) {
        this.currentRowContent = currentRowContent;
        return this;
    }

    FieldsParsingStatusBuilder withCurrentRow(int currentRow) {
        this.currentRow = currentRow;
        return this;
    }

    FieldsParsingStatusBuilder withHeaderNumberOfRowsForCurrentField(int headerNumberOfRowsForCurrentField) {
        this.headerNumberOfRowsForCurrentField = headerNumberOfRowsForCurrentField;
        return this;
    }

    FieldsParsingStatusBuilder withFieldsParsed(ArrayList<Field> fieldsParsed) {
        this.fieldsParsed = fieldsParsed;
        return this;
    }

    FieldsParsingStatus build() {
        return new FieldsParsingStatus(
                currentField,
                currentRowContent,
                currentRow,
                headerNumberOfRowsForCurrentField,
                fieldsParsed
        );
    }
}

File: minesweeper/kotlin-mockk-minesweeper/src/main/kotlin/it/chicio/minesweeper/field/parser/FieldsParsingStatusBuilder.kt
package it.chicio.minesweeper.field.parser

import it.chicio.minesweeper.field.Field
import java.util.*

class FieldsParsingStatusBuilder {
    private var currentField: Field? = null
    private var currentRowContent: String? = null
    private var currentRow = 0
    private var headerNumberOfRowsForCurrentField = 0
    private var fieldsParsed: ArrayList<Field>? = null
    fun withCurrentField(currentField: Field?): FieldsParsingStatusBuilder {
        this.currentField = currentField
        return this
    }

    fun withCurrentRowContent(currentRowContent: String?): FieldsParsingStatusBuilder {
        this.currentRowContent = currentRowContent
        return this
    }

    fun withCurrentRow(currentRow: Int): FieldsParsingStatusBuilder {
        this.currentRow = currentRow
        return this
    }

    fun withHeaderNumberOfRowsForCurrentField(headerNumberOfRowsForCurrentField: Int): FieldsParsingStatusBuilder {
        this.headerNumberOfRowsForCurrentField = headerNumberOfRowsForCurrentField
        return this
    }

    fun withFieldsParsed(fieldsParsed: ArrayList<Field>?): FieldsParsingStatusBuilder {
        this.fieldsParsed = fieldsParsed
        return this
    }

    fun build(): FieldsParsingStatus {
        return FieldsParsingStatus(
                currentField,
                currentRowContent,
                currentRow,
                headerNumberOfRowsForCurrentField,
                fieldsParsed
        )
    }
}